### PR TITLE
Use vendor guava 32.1.2-jre

### LINKED ISSUE #27801: [Task]: Upgrade vendored guava to 32.1.2-jre
### What needs to happen?

The last time we upgraded guava was back to 2019 (v20->v26) which is 4 years old now. There are also a few (temp file) vulnerabilities since then. In general these vulnerabilities are not exploitable in Beam, as these lib are only used by Beam code, we should upgrade the vendor dependency

A good candidate is the latest 32.1.2-jre, which is also included in Google Cloud Java lib LTS 5.0: https://github.com/GoogleCloudPlatform/cloud-opensource-java/pull/2343

### Issue Priority

Priority: 2 (default / most normal work should be filed as P2)

### Issue Components

- [ ] Component: Python SDK
- [x] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

### PR DESCRIPTION
**Pull request by Abacn**
Fixes #27801

The first commit is trivial 26.0-jre->32.1.2-jre replace, and the following commit(s) fix the errorprone/checkstyle rules that are introduced in new guava versions

**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by Abacn** [comment]
Run CommunityMetrics PreCommit

**Comment by Abacn** [comment]
Run GoPortable PreCommit

**Comment by Abacn** [comment]
Tests passed (except the two rerunning one, and not related to java)

there are tests results not updating to github UI, 5 of them does not even has link, but all passed:

https://ci-beam.apache.org/job/beam_PreCommit_Java_Amqp_IO_Direct_Commit/547/

https://ci-beam.apache.org/job/beam_PreCommit_Java_HBase_IO_Direct_Commit/560/

https://ci-beam.apache.org/job/beam_PreCommit_Java_Solr_IO_Direct_Commit/558/

https://ci-beam.apache.org/job/beam_PreCommit_Spotless_Commit/27951/

https://ci-beam.apache.org/job/beam_PreCommit_Website_Stage_GCS_Commit/11764/

**Comment by Abacn** [comment]
R: @bvolpato @hlteoh37

**Comment by Abacn** [comment]
Run GoPortable PreCommit

File: sdks/java/io/amazon-web-services2/src/test/java/org/apache/beam/sdk/io/aws2/kinesis/KinesisReaderCheckpointTest.java [hunk 1]
```diff
@@ -35,12 +35,10 @@
 import software.amazon.awssdk.services.kinesis.model.ShardIteratorType;
 
 public class KinesisReaderCheckpointTest {
-  /**
-   * This was generated with Beam master branch.
-   * https://github.com/apache/beam/commit/ca0787642a6b3804a742326147281c99ae8d08d2
-   */
+
+  /** This was generated manually. */
```
Comment (bvolpato) [inline sdks/java/io/amazon-web-services2/src/test/java/org/apache/beam/sdk/io/aws2/kinesis/KinesisReaderCheckpointTest.java:39]: nit Can you comment on how this was generated manually? Or you just used the failure to set as a baseline?

**Review by bvolpato — APPROVED** [review]
LGTM, thanks for pushing this Yi! It will be nice to move to a newer Guava!

File: sdks/java/io/amazon-web-services2/src/test/java/org/apache/beam/sdk/io/aws2/kinesis/KinesisReaderCheckpointTest.java (hunk 1 above)
Comment (Abacn) [reply sdks/java/io/amazon-web-services2/src/test/java/org/apache/beam/sdk/io/aws2/kinesis/KinesisReaderCheckpointTest.java:39]: Yeah, just used the error message which prints `serializedCheckpoint` to set the base line:

https://github.com/apache/beam/blob/c29485f28b34e2a5b162f5966b9a5b2d1948580b/sdks/java/io/amazon-web-services2/src/test/java/org/apache/beam/sdk/io/aws2/kinesis/KinesisReaderCheckpointTest.java#L91

**Comment by Abacn** [comment]
going to use merge instead of squash to preserve two commit history (one trivial replace, one fix)

**Comment by hlteoh37** [comment]
Thanks for push this fix through! @Abacn 
Do we know when the next Beam release is scheduled?

**Pull request merged**